tools/eslint-rules/required-modules.js

```diff
@@ -11,17 +11,16 @@ const path = require('path');
 //------------------------------------------------------------------------------
 
 module.exports = function(context) {
-  // Trim required module names
-  const requiredModules = context.options;
-  const isESM = context.parserOptions.sourceType === 'module';
-
-  const foundModules = [];
-
   // If no modules are required we don't need to check the CallExpressions
-  if (requiredModules.length === 0) {
+  if (context.options.length === 0) {
     return {};
   }
 
+  // Trim required module names
+  const requiredModule = context.options[0];
```

Comment:
Doesn't this change the original rule? It could be configured to enforce multiple modules before, now it only support a single one.

Reply:
Theoretically yes, but the usage of this rule is only for the `common` module under test directory, so it wouldn't break our linter and I think maybe it acceptable ?

https://github.com/nodejs/node/blob/56ab82e9106f87512f18a5c80be6cc0a26bda66a/test/.eslintrc.yaml#L23-L24


I'm also not sure to limit this rule for a single one module configuration, or maybe should I write a new rule(maybe named `require-top-level-common`) to force common be required before any other modules ?

Any advice ?

Reply:
I think a new rule makes sense or also change this rule name as `required-modules` is misleading.

cc: @Trott

Reply:
I'm fine with this approach + name change to `required-module` or something.

Alternatively, can we add a configuration object to the rule that allows you to specify that one of the modules must be loaded before all the others and only do this "load first" check for the module that has that option set?? It might look like this:

```yaml
  node-core/required-modules: [error, [common, assert, foo], { first: common}]
```

That might be over-engineering or it might be good flexibility to build in for the future. I'm fine with either approach.

Reply:
Yup 👍, that sounds great, will try to implement this in the next PR since it's a little over-engineering for me.